### PR TITLE
feat: clarity to wasm

### PR DESCRIPTION
**Pull request by hugocaillard**
> This PR won't be merged and I'll open a new one when getting back on this topic but I'd like to keep it around until then

We only care about this commit: 588fe0be21a13ae7e28117494250faa1f69bac5f

### CODE REVIEW

**Comment by sabbyanandan** [comment]
@hugocaillard: 🤯 👏🏽 
image

**Comment by hugocaillard** [comment]
@sabbyanandan the "-33000" is actually from the commit we don't care too much about in this PR 😅 (at least for now)

But soon!

**Comment by sabbyanandan** [comment]
Much respect to you, sir, @hugocaillard!

File: components/clarity-repl/src/repl/session.rs [hunk 1]
```diff
@@ -657,14 +662,60 @@ impl Session {
         Ok((execution, contract_identifier))
     }
 
+    fn run_wasm_and_clarity(
+        &mut self,
+        snippet: String,
+        eval_hooks: Option<Vec<&mut dyn EvalHook>>,
+        cost_track: bool,
+    ) -> Result<ExecutionResult, Vec<Diagnostic>> {
+        // @todo: we should take another approach than a regex to parse it
```
Comment (obycode) [inline components/clarity-repl/src/repl/session.rs:671]: The parser could be useful here.

Comment (obycode) [reply components/clarity-repl/src/repl/session.rs:671]: Instead though, it is probably better for the wasm runtime to provide an interface that would just allow you to pass it the snippet and run that.

**Comment by hugocaillard** [comment]
This PR won't be merged and I'll open a new one when getting back on this topic but I'd like to keep it around until then

**Comment by hugocaillard** [comment]
Closing for now. Will revisit soon, see #1155

**Pull request closed without merging**